src/Mod/Part/App/FCRepAlgoAPI_Common.h

```diff
@@ -0,0 +1,51 @@
+/***************************************************************************
+ *   Copyright (c) 2024 Eric Price (CorvusCorax) <eric.price@tuebingen.mpg.de>
+ *                                                                         *
+ *   This file is part of the FreeCAD CAx development system.              *
+ *                                                                         *
+ *   This library is free software; you can redistribute it and/or         *
+ *   modify it under the terms of the GNU Library General Public           *
+ *   License as published by the Free Software Foundation; either          *
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+/**
+  * FCRepAlgoAPI provides a wrapper for various OCCT functions.
+  */
+
+#ifndef FCREPALGOAPICOMMON_H
+#define FCREPALGOAPICOMMON_H
+#include <BRepAlgoAPI_Common.hxx>
+#include <Mod/Part/App/FCRepAlgoAPI_BooleanOperation.h>
+
+
+class FCRepAlgoAPI_Common : public FCRepAlgoAPI_BooleanOperation
+{
+public:
+
+    DEFINE_STANDARD_ALLOC
+
+  
+    //! Empty constructor
+    Standard_EXPORT FCRepAlgoAPI_Common();
+    Standard_EXPORT virtual ~FCRepAlgoAPI_Common();

```

Comment:
The destructor of the base class (BRepBuilderAPI_Command) is already virtual. In modern C++ it's not needed to explicitly add the keyword `virtual` here. Instead you should mark it with `override`.

Reply:
best code is no code - removed the unneeded destructor definition - that fixes this